frame/treasury/src/lib.rs

```diff
@@ -625,6 +791,246 @@ decl_module! {
+					let _ = T::Currency::transfer(&bounty_account, &beneficiary, payout, AllowDeath); // should not fail
+					*maybe_bounty = None;
+
+					BountyDescriptions::remove(bounty_id);
+
+					Self::deposit_event(Event::<T>::BountyClaimed(bounty_id, payout, beneficiary));
+					Ok(())
+				} else {
+					Err(Error::<T>::UnexpectedStatus.into())
+				}
+			})?;
+		}
+
+		/// Cancel an active bounty. All the funds will be send to treasury.
+		///
+		/// The dispatch origin for this call must be the curator of this bounty.
+		///
+		/// - `bounty_id`: Bounty ID to cancel.
+		#[weight = T::WeightInfo::cancel_bounty()]
+		fn cancel_bounty(origin, #[compact] bounty_id: BountyIndex) {
+			let curator = ensure_signed(origin)?;
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let bounty = maybe_bounty.as_ref().ok_or(Error::<T>::InvalidIndex)?;
+
+				match bounty.status {
+					BountyStatus::Active { expires } => {
+						let now = system::Module::<T>::block_number();
+						if expires > now {
+							// only curator can cancel unexpired bounty
+							ensure!(bounty.curator == curator, Error::<T>::RequireCurator);
+						}
+					},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				}
+
+				let bounty_account = Self::bounty_account_id(bounty_id);
+
+				BountyDescriptions::remove(bounty_id);
+
+				let balance = T::Currency::free_balance(&bounty_account);
+				let _ = T::Currency::transfer(&bounty_account, &Self::account_id(), balance, AllowDeath); // should not fail
+				*maybe_bounty = None;
+
+				Ok(())
+			})?;
+
+			Self::deposit_event(Event::<T>::BountyCanceled(bounty_id));
+		}
+
+		/// Extend the expiry time of an active bounty.
+		///
+		/// The dispatch origin for this call must be the curator of this bounty.
+		///
+		/// - `bounty_id`: Bounty ID to extend.
+		#[weight = T::WeightInfo::extend_bounty_expiry()]
+		fn extend_bounty_expiry(origin, #[compact] bounty_id: BountyIndex) {
+			let curator = ensure_signed(origin)?;
+
+			Bounties::<T>::try_mutate_exists(bounty_id, |maybe_bounty| -> DispatchResult {
+				let mut bounty = maybe_bounty.as_mut().ok_or(Error::<T>::InvalidIndex)?;
+
+				ensure!(bounty.status.is_active(), Error::<T>::UnexpectedStatus);
+				ensure!(bounty.curator == curator, Error::<T>::RequireCurator);
+
+				match bounty.status {
+					BountyStatus::Active { expires } => {
+						let expires = expires.max(system::Module::<T>::block_number() + T::BountyDuration::get());
+						bounty.status = BountyStatus::Active { expires };
+					},
+					_ => return Err(Error::<T>::UnexpectedStatus.into()),
+				}
+
+				Ok(())
+			})?;
+
+			Self::deposit_event(Event::<T>::BountyExtended(bounty_id));
+		}
+
+		/// Update `BountyValueMinimum`.
+		///
+		/// May only be called from `T::ApproveOrigin`.
+		///
+		/// - `bounty_id`: Bounty ID to extend.
+		#[weight = T::WeightInfo::update_bounty_value_minimum()]
+		fn update_bounty_value_minimum(origin, #[compact] new_value: BalanceOf<T>) {
```

Comment:
Need to check that this value is greater than ED, but better is to not make this a storage item as I suggested above